loopd/daemon.go

```diff
@@ -498,21 +499,48 @@ func (d *Daemon) initialize(withMacaroonService bool) error {
 		}
 	}
 
+	// Start the notification manager.
+	notificationCfg := &notifications.Config{
+		Client: loop_swaprpc.NewSwapServerClient(swapClient.Conn),
+	}
+	notificationManager := notifications.NewManager(notificationCfg)
+
+	d.wg.Add(1)
+	notificationInitChan := make(chan struct{})
+	go func() {
+		defer d.wg.Done()
+		err = notificationManager.Run(d.mainCtx, notificationInitChan)
+		if err != nil {
+			d.internalErrChan <- err
+		}
+	}()
+
+	select {
+	case <-notificationInitChan:
+		break
+	case <-time.After(15 * time.Second):
+		return fmt.Errorf("notification manager not ready")
+	case <-d.mainCtx.Done():
+		return fmt.Errorf("notification manager not ready: %v", d.mainCtx.Err())
```

Comment:
Is the error message correct? If this happen, the root cause is `mainCtx` cancellation, not notification manager readiness.